mmv1/products/gkeonprem/VmwareCluster.yaml

```diff
@@ -0,0 +1,673 @@
+          (default: 1 replica).
+        default_value: 1
+      - !ruby/object:Api::Type::NestedObject
+        name: autoResizeConfig
+        description: |
+          AutoResizeConfig provides auto resizing configurations.
+        default_from_api: true
+        properties:
+          - !ruby/object:Api::Type::Boolean
+            name: enabled
+            description: |
+              Whether to enable control plane node auto resizing.
+            required: true
+      - !ruby/object:Api::Type::NestedObject
+        name: vsphereConfig
+        description: Vsphere-specific config.
+        output: true
+        properties:
+          - !ruby/object:Api::Type::String
+            name: datastore
+            description: The Vsphere datastore used by the Control Plane Node.
+            output: true
+  - !ruby/object:Api::Type::NestedObject
+    name: antiAffinityGroups
+    description: |
+      AAGConfig specifies whether to spread VMware User Cluster nodes across at
+      least three physical hosts in the datacenter.
+    default_from_api: true
+    properties:
+      - !ruby/object:Api::Type::Boolean
+        name: aagConfigDisabled
+        description: |
+          Spread nodes across at least three physical hosts (requires at least three
+          hosts).
+          Enabled by default.
+        required: true
+  - !ruby/object:Api::Type::NestedObject
+    name: storage
+    description: Storage configuration.
+    default_from_api: true
+    properties:
+      - !ruby/object:Api::Type::Boolean
+        name: vsphereCsiDisabled
+        description: |
+          Whether or not to deploy vSphere CSI components in the VMware User Cluster.
+          Enabled by default.
+        required: true
+  - !ruby/object:Api::Type::NestedObject
+    name: networkConfig
+    description: The VMware User Cluster network configuration.
+    properties:
+      - !ruby/object:Api::Type::Array
+        name: "serviceAddressCidrBlocks"
+        item_type: Api::Type::String
+        required: true
+        description: |
+          All services in the cluster are assigned an RFC1918 IPv4 address 
+          from these ranges. Only a single range is supported.. This field 
+          cannot be changed after creation.
+      - !ruby/object:Api::Type::Array
+        name: "podAddressCidrBlocks"
+        item_type: Api::Type::String
+        required: true
+        description: |
+          All pods in the cluster are assigned an RFC1918 IPv4 address from these ranges. 
+          Only a single range is supported. This field cannot be changed after creation.
+      - !ruby/object:Api::Type::NestedObject
+        name: "staticIpConfig"
+        description: Configuration settings for a static IP configuration.
+        exactly_one_of:
+          - network_config.0.static_ip_config
+          - network_config.0.dhcp_ip_config
+        properties:
+          - !ruby/object:Api::Type::Array
+            name: ipBlocks
```

Comment:
Is it valid to supply to the following?

```
static_ip_config {}
```

If not, we may want to apply required and/or min_items here.

Reply:
Added required as ip_blocks must be provided. Thanks for the correction!